### ISSUE TITLE
Use Plyr for the HTML5 video player

### ISSUE THREAD

**Issue by pH-7** [pH7Software/pH7-Social-Dating-CMS#641]
It would be nice to use something like Plyr as an HTML5 media player for the ["video" module](https://github.com/pH7Software/pH7-Social-Dating-CMS/tree/master/_protected/app/system/modules/video).

https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/master/_protected/app/system/core/classes/design/VideoDesignCore.php#L91-L95

**Comment by dimuller** [comment]
Good idea! I will check it out. Thanks!

**Comment by Undefined-Variables** [comment]
Looks wonderful, going to make it happen for sure.

**Comment by Undefined-Variables** [comment]
This is not as difficult as you might think! The player already used is video.js which is the same base player that PLYr has been built upon! 

There are two choices you can do this upgrade in the correct way, this means rewriting the api class in framework/video/api and then you can use a single script to return the video information if it’s embed and another for uploading and encoding! 

This can be done like this as each services class has been standardised in PLYr. 

All that’s left to do is write the script to control the remote player! This is best done using JSON key value pairs but it this seems like hard work you can take option 2...

https://GitHub.com/spaschen/Kirby-plytag 

This was a plug-in for Kirby CMS HTML5video but works with PLYr flawlessly as it’s the exact same base video.js player used! 

Personally I’m going with option 1. Almost fallen in love with the concept behind Rest API’s and the unlimited possibilities! Postman is a great tool too! 

I will summit a pull when I have completed the script I just started! Player looks great and a huge upgrade on the most basic player I think I’ve ever seen!

**Comment by Undefined-Variables** [comment]
I have integrated Fluid_Player and it really does work well, also fully supports VAST :)

**Comment by pH-7** [comment]
> I have integrated Fluid_Player and it really does work well, also fully supports VAST :)

Really? how nice, that's great! Can't wait to see this 🤗

**Comment by Undefined-Variables** [comment]
Yeah playing the videos isn't as hard as people think. The thing I have found to course me the most problems is conversion using ffmpeg as you have to think about a lot of formats that could be uploaded but likely never will be and quality plays a big role along side slight frame rate differences and sound too. 

Think I have found a nice balance encoding with libx245 to h245 in 480, 720, 1080 and libfacc @ 96k. But 90% of video is 25fps but the odd device records at 30fps and this is a real headache for ffmpeg as its processed and thinks it was successful but in reality its jammed in the conversion queue stopping the process.

Once its perfect I will pull request :)

**Comment by Undefined-Variables** [comment]
1080 is not an exact science yet either its temperamental!

**Comment by pH-7** [comment]
Thanks for the suggestion.

I reviewed this during backlog cleanup. This request is valid but outside the current 18.x delivery scope, so I’m closing it as not planned for now to keep the tracker focused on active work.

If you want to push this forward, please open a new proposal with a minimal implementation plan (scope, DB/API impact, migration notes, and backward-compatibility expectations).

Take care and thank you for using pH7 Social Dating Builder,

[Pierre-Henry](https://github.com/pH-7)
